servers/src/main/java/tachyon/master/file/FileSystemMaster.java

```diff
@@ -569,6 +617,33 @@ public long createFile(TachyonURI path, long blockSizeBytes, boolean recursive,
     if (mWhitelist.inList(path.toString())) {
       inode.setCacheable(true);
     }
+
+    if (ttl != Constants.NO_TTL) {
+      long ttlEnd = opTimeMs + ttl;
+      int ttlBucketIndex = Math.min((int) (ttlEnd / mTTLCheckerIntervalMs), mTTLBuckets.size() - 1);
+      if (ttlBucketIndex < 0) {
+        // There is no existing bucket, add a new bucket, the interval this new bucket is the
+        // interval where ttlEnd resides in.
+        mTTLBuckets.add(new TTLBucket(ttlBucketIndex * mTTLCheckerIntervalMs));
+        ttlBucketIndex = 0;
+      }
+
+      TTLBucket bucket = mTTLBuckets.get(ttlBucketIndex);
```

Comment:
I will add more documentation to a new class TTLBuckets with all related operation aggregated.